### PR TITLE
upkeep/57: bump min WP and PHP versions

### LINKED ISSUE #57: Bump WordPress and PHP minimums
Bump WP & PHP Minimums to the following
 
- [ ] WordPress to 5.7
- [ ] PHP to 7.4
- [ ] Remove any no-longer-needed conditional code for older version support.
- [ ] Take a look at adjusting our testing matrices, if applicable.
- [ ] Bump the minimums in plugin documentation and header fields.

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

Closes #57

### Changelog Entry

> Changed - Bump minimum PHP version from 5.6 to 7.4
> Changed - Bump minimum node version from 12 to 16
> Changed - Bump minimum composer version to 2

### Credits

Props @vikrampm1 @Sidsector9 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v16
```
Comment (cadic) [inline .nvmrc:1]: The [eslint.yml](https://github.com/10up/retro-winamp-block/blob/59ef1614ecae2119c5b89fbd0d6183d277ed7e8f/.github/workflows/eslint.yml#L21) workflow is using hardcoded Node version, should we refer to this `.nvmrc` file instead?

**Review by cadic — CHANGES_REQUESTED** [review]
All good, maybe use .nvmrc in existing workflow?

File: .nvmrc (hunk 1 above)
Comment (Sidsector9) [reply .nvmrc:1]: @cadic this is done!

**Review by cadic — APPROVED** [review]
LGTM!

I see WPCS action failure, not related to the current change

**Pull request merged**